src/ch/ricardo/plugins/intellij/mockery/action/GenMockeryActionHandler.java

```diff
@@ -98,10 +98,36 @@ public void doExecute(@NotNull Editor editor, Caret caret, DataContext dataConte
 
         // Do the generation
         Path filePath = Paths.get(FileDocumentManager.getInstance().getFile(editor.getDocument()).getCanonicalPath());
-
         GeneralCommandLine commandLine = PtyCommandLine.isEnabled() ? new PtyCommandLine() : new GeneralCommandLine();
         commandLine.withWorkDirectory(filePath.getParent().toString());
-        commandLine.setExePath(EnvironmentUtil.getValue("GOPATH") + "/bin/mockery");
+
+        String[] envGoPaths = EnvironmentUtil.getValue("GOPATH").split(File.pathSeparator);
+        String mockeryPath = "";
+
+        for (String goPath : envGoPaths) {
+            File goPathBin = new File(goPath + "/bin");
+            if(!goPathBin.exists() || !goPathBin.isDirectory())
+                continue;
+
+            String[] goPathBinFiles = goPathBin.list();
+            if(goPathBinFiles == null)
+                continue;
+
+            for(String goPathBinExecutable : goPathBinFiles)
+                if(goPathBinExecutable.startsWith("mockery.")) {
```

Comment:
You right.
I'm wondering if 'startsWith("mockery")' should be enough, or maybe I should include the OS extension in the file searching.

Reply:
Having the OS extension would be great as it would allow for doing a "equals" rather than "startsWith" which makes the code more robust.

Anyways, no need to over-engineer here: a simple .equals("mockery") || .equals("mockery.exe") would be already a good balance between robustness and effort.